app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CategoriesController < ApplicationController
+  before_action :find_category, only: [:show, :edit, :update]
+
+  def index
+    @categories = Category.all
+    @category = Category.new
+  end
+
+
+  def show
+    @products = @category.products.paginate(page: params[:page], per_page: 12).where(active: true)
+  end
+
+  def edit
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      redirect_to categories_path
+    else
+      @categories = Category.all
+      render :index
+    end
+  end
+
+# Removing this functionality. Maybe add it back in for admin user
```

Comment:
Commented out code like this should generally be deleted entirely. Git will retain a history of our codebase, so if it existed previously in un-commented form, we'll still be able to find it later should we need to.